modules/ezcontent_node/modules/ezcontent_smart_article/src/Form/ConfigForm.php

```diff
@@ -195,6 +195,53 @@ public function buildForm(array $form, FormStateInterface $form_state) {
         'visible' => ['select[name="image_tagging_service"]' => ['value' => 'srijan_image_tagging']],
       ],
     ];
+    $form['gcm_secret_key_image_tags'] = [
+      '#type' => 'textfield',
+      '#title' => $this->t('Enter Secret Key'),
+      '#description' => $this->t('Provide the google vision api secret key to generate image tags.'),
+      '#default_value' => $config->get('gcm_secret_key_image_tags'),
+      '#states' => [
+        'visible' => ['select[name="image_tagging_service"]' => ['value' => 'google_image_tagging']],
+      ],
+    ];
```

Comment:
Where are we checking if the GCP class exists or not ? if we are not including the SDK, wont it throw errors ?

Reply:
For GCP we don't have sdk , we have endpoint which being hit with secret key.If secret key is not there the ajax request will fail and tag won't generate.It won't cause any fatal error.